hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestIPC.java

```diff
@@ -1728,6 +1728,55 @@ public void testProxyUserBinding() throws Exception {
     checkUserBinding(true);
   }
 
+  @Test(timeout=60000)
+  public void testUpdateAddressEnsureResolved() throws Exception {
+    // start server
+    Server server = new TestServer(5, false);
+    server.start();
+
+    SocketFactory mockFactory = Mockito.mock(SocketFactory.class);
+    doThrow(new ConnectTimeoutException("fake")).when(mockFactory).createSocket();
+    Client client = new Client(LongWritable.class, conf, mockFactory);
+    InetSocketAddress address = new InetSocketAddress("localhost", 9090);
```

Comment:
Does 9090 has any relevance here? If the port is occupied in the env where test run, will there be any impact? If yes, then use ```NetUtils.getFreeSocketPort()```